### PR TITLE
Added Digital Ocean integration to the API

### PR DESCRIPTION
**Pull request by nghiapham1026**

### CODE REVIEW

File: apps/web/hooks/useBucket.tsx [hunk 1]
```diff
@@ -35,6 +35,8 @@ export default function useBucket(): ContextValue {
 		return useS3();
 	} else if ((Provider[keys.type] as Provider) === Provider.wasabi) {
 		return useS3();
+	} else if ((Provider[keys.type] as Provider) === Provider.digitalocean) {
+		return useS3();
```
Comment (sabrinaspage) [inline apps/web/hooks/useBucket.tsx:39]: Shall we consider turning this into a switch statement?
https://github.com/nghiapham1026/firefiles/blob/3650bf44eaf6fc21eb2acb5062d6bc277870f180/apps/web/hooks/useBucket.tsx#L28-L40
Something along the lines of:
```
// store the type of key as a variable to avoid repetition
providerKey = (Provider[keys.type] as Provider;

// view this example https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#using_switch
switch (providerKey):
   case Provider.s3:
   case Provider.backblaze:
   case Provider.wasabi:
   case Provider.digitalocean:
      return useS3();
   case Provider.firebase:
      return useFirebase();
   case default:
      return null;
```

File: apps/web/hooks/useS3.tsx [hunk 2]
```diff
@@ -69,6 +69,8 @@ export const S3Provider: React.FC<PropsWithChildren<Props>> = ({
 			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.backblazeb2.com`;
 		} else if ((Provider[data.type] as Provider) === Provider.wasabi) {
 			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.wasabisys.com`;
+		} else if ((Provider[data.type] as Provider) === Provider.digitalocean) {
+			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.digitalocean.com`;
```
Comment (sabrinaspage) [inline apps/web/hooks/useS3.tsx:73]: Same here; we can create a helper function that allows us to get the bucket url:
```
const getBucketUrl = (data) => {
  const provider = Provider[data.type];

  switch (provider) {
    case Provider.s3:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.amazonaws.com`;
    case Provider.backblaze:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.backblazeb2.com`;
    case Provider.wasabi:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.wasabisys.com`;
    case Provider.digitalocean:
      return `https://${data.keys.Bucket}.s3.${data.keys.region}.digitalocean.com`;
    default:
      return null; // or throw an error if necessary
  }
};

data.keys.bucketUrl = getBucketUrl(data);
```

We apply separation of concerns by having one line handle assignment and one function determine the bucket URL.

Using a switch statement is easy to expand upon as well.

**Comment by sabrinaspage** [comment]
pinging @9brian so he has visibility into my comments

File: apps/web/pages/new/wasabiProvider.tsx
Comment (sabrinaspage) [inline apps/web/pages/new/wasabiProvider.tsx:1]: why was this file deleted?

File: apps/web/hooks/useS3.tsx (hunk 2 above)
Comment (sabrinaspage) [reply apps/web/hooks/useS3.tsx:73]: Sometimes it is better to leave a repo better than it was before, which is why this is a suggestion.

File: apps/web/pages/new/wasabiProvider.tsx
Comment (9brian) [reply apps/web/pages/new/wasabiProvider.tsx:1]: The file was deleted because we didnt need to create a new Provider. As per the issue, we could use the useS3 hook to integrate wasabi https://github.com/faisalsayed10/firefiles/issues/39

Comment (sabrinaspage) [reply apps/web/pages/new/wasabiProvider.tsx:1]: Thanks for the insight!

File: apps/web/pages/new/digitalocean.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+import axios from "axios";
+import Head from "next/head";
+import { useRouter } from "next/router";
+import { useState } from "react";
+import toast from "react-hot-toast";
+import { ArrowNarrowLeft } from "tabler-icons-react";
+import "video-react/dist/video-react.css";
+import validator from "validator";
+
+const NewS3 = () => {
```
Comment (sabrinaspage) [inline apps/web/pages/new/digitalocean.tsx:25]: This class follows the structure of other classes in this directory. Can we leverage a parent class (aka a base class) with inheritable methods so we do not need to repeat the same patterns?

Just a thought to chew on.
